management/server/setupkey.go

```diff
@@ -279,6 +283,16 @@ func (am *DefaultAccountManager) SaveSetupKey(ctx context.Context, accountID str
 		return nil, status.Errorf(status.NotFound, "setup key not found")
 	}
 
+	for _, group := range keyToSave.AutoGroups {
+		g, ok := account.Groups[group]
```

Comment:
let's make these two checks a function check for both, create and save setup key